src/zip_code_tree.hpp

```diff
@@ -405,114 +426,231 @@ class ZipCodeForest {
+        }
+
+    };
+
+
+    // For children of snarls, we need to remember the siblings and start bound that came before 
+    // them so we can record their distances
+    // This holds the indices (into zip_code_tree) of each seed or start of a chain,
+    //  and each start and child chain start of a snarl
+    // For the children of a chain, the value is the prefix sum in the chain (relative to the 
+    // orientation of the top-level chain, not necessarily the chain itself)
+    // For the children of a snarl, the value is the index of the CHAIN_START in zip_code_tree.
     //  The first seed in the chain will need to be found by looping through zip_code_tree
     struct child_info_t {
+
         ZipCodeTree::tree_item_type_t type;  //the type of the item
-        size_t value;  //A value associated with the item, either the offset in a chain, index of the snarl child start
+
+        //A value associated with the item, either the offset in a chain, index of the snarl 
+        //child start
+        size_t value;  
     
-        //For the children of snarls, the distance to the left and right of the chain, that gets added to
-        //edges in the snarl
+        //For the children of snarls, the distance to the left and right of the chain, that gets 
+        //added to edges in the snarl
         std::pair<size_t, size_t> distances;
 
         //Is the sibling reversed. 
-        //This is only used for children of snarls, to indicate that the child is traversed backwards 
+        //This is only used for children of snarls, to indicate that the child is traversed 
+        //backwards 
         bool is_reversed = false;
     };
 
+    struct interval_and_orientation_t {
+
+        //Indices into zipcode_sort_order
+        size_t interval_start : 26; //inclusive
+        size_t interval_end : 26;   //exclusive
+
+        // is_reversed is true if that snarl tree node is reversed relative to the 
+        // top-level chain
+        bool is_reversed : 1;
+
+        //The type of the snarl tree structure.
+        // For nodes on chains, all seeds on the chain that aren't nested in snarls are put in 
+        // the same interval, regardless of if they are actually on the same node
+        ZipCode::code_type_t code_type : 5;
+
+        size_t depth : 14;
+
+        //For children of cyclic snarls, an entire chain may be duplicated in the opposite 
+        // orientation immediately after the first copy. In this case, when the second copy is 
+        // processed, the entire interval is already in the correct order, just reversed. 
+        //If this is_reverse_ordered true, then the interval is sorted in the reverse order, so it
+        // needs to be flipped before processing
+        bool is_reverse_ordered;
+
+        //After flipping a reverse-ordered interval, all of the child intervals will be sorted
+        //So remember if the interval doesn't need sorting
+        bool is_ordered;
```

Comment:
If you really want to bit pack this maybe the flags should be together?

Also, I'm a little worried this type is getting away from being what its name says it is (a pair of an interval and an orientation?) and I kind of don't understand what it really is. It looks like maybe it is some kind of scratch space for doing bookkeeping about what range is covered by part of the in-progress zip tree and also how the things are laid out in that space and how they are ordered. Maybe it's really an `interval_state` or `interval_metadata` something.